Remove `cfg(doc)` from std::os module reexports to fix rustdoc linking issues

GuillaumeGomez opened the pull request:
Fixes https://github.com/rust-lang/rust/issues/88304.

I tested it based on https://github.com/rust-lang/rust/pull/88292.

Not sure if it's the best approach, but at least it makes thing a bit simpler.

cc @jyn514

rust-highfive commented:
r? @joshtriplett

(rust-highfive has picked a reviewer for you, use r? to override)

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/3507575444) [(plain)](https://github.com/rust-lang/rust/commit/5e090fdbe88c31ab54cc761a4ed57eaed92fd4e0/checks/3507575444/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
configure: rust.channel         := nightly
configure: rust.debug-assertions := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
configure: 
```

</details>

jyn514 commented:
I don't think this is the right approach. We should find why the link-to-definition pass is broken and fix it, not change the standard library to avoid rustdoc bugs.

GuillaumeGomez commented:
Well, it's not specific to the jump to definition feature. The link itself is correct after all... The problem is that we have a different rendering with "cfg(doc)". We literally can't fix it.

jyn514 commented:
Got it, that makes sense. This seems like a reasonable change then, to make the information available to rustdoc when documenting it as a dependency.

GuillaumeGomez commented:
I'll check what's wrong with the changes I made.

GuillaumeGomez commented:
So: in the first commit I used `cfg-if` to have a clean state to work on. But because of https://github.com/rust-analyzer/rust-analyzer/issues/6038, I removed it in the second commit and reverted back to code without macro.

GuillaumeGomez commented:
Seems like it's ready for review! :)

jyn514 reviewed (commented):
@GuillaumeGomez I was planning to let @joshtriplett review, I don't think we should be changing fundamental things about libstd without asking T-libs.

GuillaumeGomez commented:
> @GuillaumeGomez I was planning to let @joshtriplett review, I don't think we should be changing fundamental things about libstd without asking T-libs.

Which is why I didn't set you as reviewer. ;)

But reviews from you remain very welcome!

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/3534369810) [(plain)](https://github.com/rust-lang/rust/commit/0c1e11af198defc0972a4142f46460e923dc42b8/checks/3534369810/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
configure: rust.channel         := nightly
configure: rust.debug-assertions := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
configure: 
---
skip untracked path cpu-usage.csv during rustfmt invocations
skip untracked path src/doc/book/ during rustfmt invocations
skip untracked path src/doc/rust-by-example/ during rustfmt invocations
skip untracked path src/llvm-project/ during rustfmt invocations
Diff in /checkout/library/std/src/os/mod.rs at line 100:
 pub mod emscripten;
 #[cfg(target_os = "espidf")]
 pub mod espidf;
+#[cfg(all(target_vendor = "fortanix", target_env = "sgx"))]
+pub mod fortanix_sgx;
 #[cfg(target_os = "freebsd")]
 pub mod freebsd;
 #[cfg(target_os = "fuchsia")]
Diff in /checkout/library/std/src/os/mod.rs at line 106:
 pub mod fuchsia;
-#[cfg(all(target_vendor = "fortanix", target_env = "sgx"))]
-pub mod fortanix_sgx;
 #[cfg(target_os = "haiku")]
 pub mod haiku;
 #[cfg(target_os = "illumos")]
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/rustfmt" "--config-path" "/checkout" "--edition" "2018" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/os/wasi/io/mod.rs" "/checkout/library/std/src/os/windows/io/raw.rs" "/checkout/library/std/src/os/wasi/io/raw.rs" "/checkout/library/std/src/os/mod.rs" "/checkout/library/std/src/os/fortanix_sgx/arch.rs" "/checkout/library/std/src/os/fd/owned.rs" "/checkout/library/std/src/os/fortanix_sgx/mod.rs" "/checkout/library/std/src/os/wasi/io/fd/tests.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
```

</details>

GuillaumeGomez commented:
r? @Amanieu

Amanieu commented on library/std/src/os/mod.rs:
```diff
@@ -10,117 +10,121 @@ pub mod raw;
+    all(target_arch = "wasm32", not(target_os = "wasi")),
+    all(target_vendor = "fortanix", target_env = "sgx")
+)))]
+#[cfg(target_os = "hermit")]
```
```suggestion
#[cfg(all(not(doc), target_os = "hermit"))]
```
Otherwise `hermit` doc builds will have 2 unix modules.

GuillaumeGomez commented:
Updated!

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit 88d5aa62dc6e30e6b48e2c280e914752ea26937e has been approved by `Amanieu`

<!-- @bors r=Amanieu 88d5aa62dc6e30e6b48e2c280e914752ea26937e -->
<!-- homu: {"type":"Approved","sha":"88d5aa62dc6e30e6b48e2c280e914752ea26937e","approver":"Amanieu"} -->

bors commented:
:evergreen_tree: The tree is currently [closed](https://github.com/rust-lang/rust/pull/88783#issuecomment-916338297) for pull requests below priority 100. This pull request will be tested once the tree is reopened.

GuillaumeGomez commented:
Seems like `wasm32` is picking `unix` for whatever reason...

Amanieu commented:
`wasm32-emscripten` is unix, `wasm32-wasi` and `wasm32-unknown` aren't.

GuillaumeGomez commented:
Interesting, didn't know that. I'll update then.

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/3576551738) [(plain)](https://github.com/rust-lang/rust/commit/a49b41adcc60e37f0fa13f212823393d36c52966/checks/3576551738/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
configure: rust.channel         := nightly
configure: rust.debug-assertions := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
configure: 
---
skip untracked path src/llvm-project/ during rustfmt invocations
Diff in /checkout/library/std/src/os/mod.rs at line 53:
 pub mod windows {}
 
 // unix
-#[cfg(not(all(doc, any(
-    all(target_arch = "wasm32", not(target_os = "wasi")),
-    all(target_vendor = "fortanix", target_env = "sgx")
-))))]
+#[cfg(not(all(
+    doc,
+    any(
+        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_vendor = "fortanix", target_env = "sgx")
+)))]
+)))]
 #[cfg(target_os = "hermit")]
 #[path = "hermit/mod.rs"]
 pub mod unix;
Diff in /checkout/library/std/src/os/mod.rs at line 63:
-#[cfg(not(all(doc, any(
-    all(target_arch = "wasm32", not(target_os = "wasi")),
-    all(target_vendor = "fortanix", target_env = "sgx")
-))))]
+#[cfg(not(all(
+    doc,
+    any(
+        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_vendor = "fortanix", target_env = "sgx")
+)))]
+)))]
 #[cfg(all(not(target_os = "hermit"), any(unix, doc)))]
 pub mod unix;
Diff in /checkout/library/std/src/os/mod.rs at line 70:
 // linux
 // linux
-#[cfg(not(all(doc, any(
-    all(target_arch = "wasm32", not(target_os = "wasi")),
-    all(target_vendor = "fortanix", target_env = "sgx")
-))))]
+#[cfg(not(all(
+    doc,
+    any(
+        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_vendor = "fortanix", target_env = "sgx")
+)))]
+)))]
 #[cfg(any(target_os = "linux", target_os = "l4re", doc))]
 pub mod linux;
Diff in /checkout/library/std/src/os/mod.rs at line 78:
 // wasi
 // wasi
-#[cfg(not(all(doc, any(
-    all(target_arch = "wasm32", not(target_os = "wasi")),
-    all(target_vendor = "fortanix", target_env = "sgx")
-))))]
+#[cfg(not(all(
+    doc,
+    any(
+        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_vendor = "fortanix", target_env = "sgx")
+)))]
+)))]
 #[cfg(any(target_os = "wasi", doc))]
 pub mod wasi;
Diff in /checkout/library/std/src/os/mod.rs at line 86:
 // windows
 // windows
-#[cfg(not(all(doc, any(
-    all(target_arch = "wasm32", not(target_os = "wasi")),
-    all(target_vendor = "fortanix", target_env = "sgx")
-))))]
+#[cfg(not(all(
+    doc,
+    any(
+        all(target_arch = "wasm32", not(target_os = "wasi")),
+        all(target_vendor = "fortanix", target_env = "sgx")
+)))]
+)))]
 #[cfg(any(windows, doc))]
 pub mod windows;
 
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/rustfmt" "--config-path" "/checkout" "--edition" "2018" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/os/fd/net.rs" "/checkout/library/std/src/os/fortanix_sgx/io.rs" "/checkout/library/std/src/os/fd/owned.rs" "/checkout/library/std/src/os/fortanix_sgx/ffi.rs" "/checkout/library/std/src/os/fd/mod.rs" "/checkout/library/std/src/os/fd/raw.rs" "/checkout/library/std/src/os/netbsd/raw.rs" "/checkout/library/std/src/os/mod.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
```

</details>

Amanieu commented:
@bors r+

bors commented:
:pushpin: Commit b98ba6f190c219b79de07477b0c6d74072678799 has been approved by `Amanieu`

<!-- @bors r=Amanieu b98ba6f190c219b79de07477b0c6d74072678799 -->
<!-- homu: {"type":"Approved","sha":"b98ba6f190c219b79de07477b0c6d74072678799","approver":"Amanieu"} -->

bors commented:
:hourglass: Testing commit b98ba6f190c219b79de07477b0c6d74072678799 with merge 39cc9ca3814d80e7488acfce2ddc21f04e7a0aa3...
<!-- homu: {"type":"BuildStarted","head_sha":"b98ba6f190c219b79de07477b0c6d74072678799","merge_sha":"39cc9ca3814d80e7488acfce2ddc21f04e7a0aa3"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/3584913795) [(plain)](https://github.com/rust-lang-ci/rust/commit/39cc9ca3814d80e7488acfce2ddc21f04e7a0aa3/checks/3584913795/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

 error: using a non-octal value to set unix file permissions
   --> $DIR/non_octal_unix_permissions.rs:12:18
    |
 LL |     options.mode(440);
    |                  ^^^ help: consider using an octal literal instead: `0o440`
    |
    = note: `-D clippy::non-octal-unix-permissions` implied by `-D warnings`
 error: using a non-octal value to set unix file permissions
-  --> $DIR/non_octal_unix_permissions.rs:17:47
-   |
-   |
-LL |     let _permissions = Permissions::from_mode(647);
-   |                                               ^^^ help: consider using an octal literal instead: `0o647`
-error: using a non-octal value to set unix file permissions
   --> $DIR/non_octal_unix_permissions.rs:26:26
    |
 LL |     permissions.set_mode(644);
 LL |     permissions.set_mode(644);
    |                          ^^^ help: consider using an octal literal instead: `0o644`
 error: using a non-octal value to set unix file permissions
   --> $DIR/non_octal_unix_permissions.rs:31:18
    |
    |
 LL |     builder.mode(755);
    |                  ^^^ help: consider using an octal literal instead: `0o755`
-error: aborting due to 4 previous errors
+error: aborting due to 3 previous errors
 
 
 

The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/test_build_base/non_octal_unix_permissions.stage-id.stderr
diff of fixed:

 // ignore-windows
 // run-rustfix
 #![warn(clippy::non_octal_unix_permissions)]
 use std::fs::{DirBuilder, File, OpenOptions, Permissions};
 use std::os::unix::fs::{DirBuilderExt, OpenOptionsExt, PermissionsExt};
 fn main() {
     let permissions = 0o760;
 
     // OpenOptionsExt::mode
     // OpenOptionsExt::mode
     let mut options = OpenOptions::new();
     options.mode(0o440);
     options.mode(0o400);
     options.mode(permissions);
 
     // PermissionsExt::from_mode
-    let _permissions = Permissions::from_mode(0o647);
+    let _permissions = Permissions::from_mode(647);
     let _permissions = Permissions::from_mode(0o000);
     let _permissions = Permissions::from_mode(permissions);
     // PermissionsExt::set_mode
     // PermissionsExt::set_mode
     let f = File::create("foo.txt").unwrap();
     let metadata = f.metadata().unwrap();
     let mut permissions = metadata.permissions();
     permissions.set_mode(0o644);
     permissions.set_mode(0o704);
 
     // DirBuilderExt::mode
     // DirBuilderExt::mode
     let mut builder = DirBuilder::new();
     builder.mode(0o755);
     builder.mode(0o406);
 

The actual fixed differed from the expected fixed.
The actual fixed differed from the expected fixed.
Actual fixed saved to /checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/test_build_base/non_octal_unix_permissions.stage-id.fixed
To only update this specific test, also pass `--test-args non_octal_unix_permissions.rs`

error: 2 errors occurred comparing output.
status: exit status: 1
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools-bin/clippy-driver" "tests/ui/non_octal_unix_permissions.rs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/test_build_base" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/test_build_base/non_octal_unix_permissions.stage-id" "-A" "unused" "--emit=metadata" "-L" "dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps" "-L" "dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps" "-Dwarnings" "-Zui-testing" "--extern" "derive_new=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps/libderive_new-ea1db0c41efed0d6.so" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libserde-89222f6e0d369b36.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libquote-c6bd037ba33baa25.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libsyn-2c42e7af51424ce5.rlib" "--extern" "if_chain=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libif_chain-b3e76e8f62643cc6.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libitertools-b0f19a5fad83a10b.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libregex-392b40a2759b0504.rlib" "--extern" "serde_derive=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps/libserde_derive-8bd56d0234290b82.so" "--extern" "clippy_utils=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/deps/libclippy_utils-bc2b0c8a9dcb9fb3.rlib" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/test_build_base/non_octal_unix_permissions.stage-id.aux"
------------------------------------------

------------------------------------------
stderr:
stderr:
------------------------------------------
{"message":"using a non-octal value to set unix file permissions","code":{"code":"clippy::non_octal_unix_permissions","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/non_octal_unix_permissions.rs","byte_start":339,"byte_end":342,"line_start":12,"line_end":12,"column_start":18,"column_end":21,"is_primary":true,"text":[{"text":"    options.mode(440);","highlight_start":18,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`-D clippy::non-octal-unix-permissions` implied by `-D warnings`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"consider using an octal literal instead","code":null,"level":"help","spans":[{"file_name":"tests/ui/non_octal_unix_permissions.rs","byte_start":339,"byte_end":342,"line_start":12,"line_end":12,"column_start":18,"column_end":21,"is_primary":true,"text":[{"text":"    options.mode(440);","highlight_start":18,"highlight_end":21}],"label":null,"suggested_replacement":"0o440","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: using a non-octal value to set unix file permissions\n  --> tests/ui/non_octal_unix_permissions.rs:12:18\n   |\nLL |     options.mode(440);\n   |                  ^^^ help: consider using an octal literal instead: `0o440`\n   |\n   = note: `-D clippy::non-octal-unix-permissions` implied by `-D warnings`\n\n"}
{"message":"using a non-octal value to set unix file permissions","code":{"code":"clippy::non_octal_unix_permissions","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/non_octal_unix_permissions.rs","byte_start":798,"byte_end":801,"line_start":26,"line_end":26,"column_start":26,"column_end":29,"is_primary":true,"text":[{"text":"    permissions.set_mode(644);","highlight_start":26,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider using an octal literal instead","code":null,"level":"help","spans":[{"file_name":"tests/ui/non_octal_unix_permissions.rs","byte_start":798,"byte_end":801,"line_start":26,"line_end":26,"column_start":26,"column_end":29,"is_primary":true,"text":[{"text":"    permissions.set_mode(644);","highlight_start":26,"highlight_end":29}],"label":null,"suggested_replacement":"0o644","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: using a non-octal value to set unix file permissions\n  --> tests/ui/non_octal_unix_permissions.rs:26:26\n   |\nLL |     permissions.set_mode(644);\n   |                          ^^^ help: consider using an octal literal instead: `0o644`\n\n"}
{"message":"using a non-octal value to set unix file permissions","code":{"code":"clippy::non_octal_unix_permissions","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/non_octal_unix_permissions.rs","byte_start":923,"byte_end":926,"line_start":31,"line_end":31,"column_start":18,"column_end":21,"is_primary":true,"text":[{"text":"    builder.mode(755);","highlight_start":18,"highlight_end":21}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"consider using an octal literal instead","code":null,"level":"help","spans":[{"file_name":"tests/ui/non_octal_unix_permissions.rs","byte_start":923,"byte_end":926,"line_start":31,"line_end":31,"column_start":18,"column_end":21,"is_primary":true,"text":[{"text":"    builder.mode(755);","highlight_start":18,"highlight_end":21}],"label":null,"suggested_replacement":"0o755","suggestion_applicability":"MachineApplicable","expansion":null}],"children":[],"rendered":null}],"rendered":"error: using a non-octal value to set unix file permissions\n  --> tests/ui/non_octal_unix_permissions.rs:31:18\n   |\nLL |     builder.mode(755);\n   |                  ^^^ help: consider using an octal literal instead: `0o755`\n\n"}

------------------------------------------

thread 'compile_test' panicked at 'Some tests failed', /cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.7.0/src/lib.rs:105:22
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/3585365449)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/3585365449","builder_name":"checks-actions"} -->

GuillaumeGomez commented:
@matthiaskrgr: Any idea what's going on with the clippy lint by any chance?

camsteffen commented:
That lint uses the absolute path of `Permissions` [here](https://github.com/rust-lang/rust-clippy/blob/92ca25b3ec8cef291f1d3b37bfd01d3cce8609e2/clippy_lints/src/non_octal_unix_permissions.rs#L52) and I'm guessing the path is changed. You can change the path in Clippy. The proper fix is to use a diagnostic item.

GuillaumeGomez commented:
Oh I see, thanks!

GuillaumeGomez commented:
@bors: r=Amanieu

bors commented:
:pushpin: Commit 9a3b1cfeb08db360b11dd313ab2907d3e8c45bb3 has been approved by `Amanieu`

<!-- @bors r=Amanieu 9a3b1cfeb08db360b11dd313ab2907d3e8c45bb3 -->
<!-- homu: {"type":"Approved","sha":"9a3b1cfeb08db360b11dd313ab2907d3e8c45bb3","approver":"Amanieu"} -->

bors commented:
:hourglass: Testing commit 9a3b1cfeb08db360b11dd313ab2907d3e8c45bb3 with merge e846f9c44f1e17cac66d7b75e2ca099c87a2dfcb...
<!-- homu: {"type":"BuildStarted","head_sha":"9a3b1cfeb08db360b11dd313ab2907d3e8c45bb3","merge_sha":"e846f9c44f1e17cac66d7b75e2ca099c87a2dfcb"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/3610056196)
Approved by: Amanieu
Pushing e846f9c44f1e17cac66d7b75e2ca099c87a2dfcb to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Amanieu","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/3610056196"},"merge_sha":"e846f9c44f1e17cac66d7b75e2ca099c87a2dfcb"} -->

rust-timer commented:
Finished benchmarking commit (e846f9c44f1e17cac66d7b75e2ca099c87a2dfcb): [comparison url](https://perf.rust-lang.org/compare.html?start=cdeba02ff71416e014f7130f75166890688be986&end=e846f9c44f1e17cac66d7b75e2ca099c87a2dfcb).

**Summary**: This benchmark run did not return any relevant changes.

If you disagree with this performance assessment, please file an issue in [rust-lang/rustc-perf](https://github.com/rust-lang/rustc-perf/issues/new).

 

@rustbot label: -perf-regression
